src/init.rs

```diff
@@ -11,29 +12,46 @@ rest of the script, and you have to spam semicolons everywhere. By using
 source and process substitutions, we make it possible to comment and debug
 the init scripts. */
 
+fn get_starship_path() -> io::Result<String> {
+    let current_exe = env::current_exe()?
+        .to_str()
+        .ok_or_else(|| io::Error::new(io::ErrorKind::Other, "can't convert to str"))?
+        .to_string();
+    Ok(current_exe)
+}
+
 /* This prints the setup stub, the short piece of code which sets up the main
 init code. The stub produces the main init script, then evaluates it with
 `source` and process substitution */
-pub fn init_stub(shell_name: &str) {
+pub fn init_stub(shell_name: &str) -> io::Result<()> {
     log::debug!("Shell name: {}", shell_name);
 
     let shell_basename = Path::new(shell_name).file_stem().and_then(OsStr::to_str);
 
+    // let starship = get_starship_path()?.replace("\"", "\"'\"'\"");
+    let starship = get_starship_path()?.replace("\"", "\"\\\"\"");
```

Comment:
shellcheck whines about this one but not `.replace("\"", "\"'\"'\"")`.

Reply:
Can we do this with raw string literals? Would make it a lot easier to read imo, especially if you have syntax highlighting.

```suggestion
    let starship = get_starship_path()?.replace(r#"""#,r#""\"""#);
```

...okay, maybe "easier" is the wrong word, but you don't have to work as hard parsing the backslashes. I'm not even sure if the suggestion is correct, as it stands right now.

Reply:
On second thought, I don't have a clear opinion on which one is better: one forces you to parse escape sequences in your head, one makes it impossible to visually separate adjacent double-quotes. Pick the one you think is better.